From a7866996b69de9dc2d2280b337a230892c7e9a11 Mon Sep 17 00:00:00 2001
From: Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
Date: Fri, 27 Jun 2014 18:53:30 +0000
Subject: [PATCH] Link KF5XmlGui to plasmashell

Needed for KActionCollection.
---
 shell/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/shell/CMakeLists.txt b/shell/CMakeLists.txt
index 0312c40..7814dc4 100644
--- a/shell/CMakeLists.txt
+++ b/shell/CMakeLists.txt
@@ -22,6 +22,7 @@ find_package(KF5Activities REQUIRED)
 find_package(KF5TextEditor) # not part of KF5.0
 find_package(KF5DBusAddons ${KF5_VERSION} REQUIRED)
 find_package(KF5Declarative ${KF5_VERSION} REQUIRED)
+find_package(KF5XmlGui ${KF5_VERSION} REQUIRED)
 
 set(HAVE_KTEXTEDITOR ${KF5TextEditor_FOUND})
 
@@ -91,6 +92,7 @@ target_link_libraries(plasmashell
  KF5::DBusAddons
  KF5::Screen
  KF5::Declarative
+ KF5::XmlGui
 )
 if (TARGET KF5::TextEditor)
  target_link_libraries(plasmashell KF5::TextEditor)
-- 
1.8.3-rc3

